### PR TITLE
Error codes

### PR DESCRIPTION
**Pull request by abarichello**
Adds error codes to success false responses, ideally we can treat those client side and display more useful error messages instead of a generic auth error

### CODE REVIEW

File: client/clientauth.js [hunk 1]
```diff
@@ -1,186 +1,194 @@
...
+		async ( request ) =>
 		{
+		// only do this if we're in an environment that actually requires session tokens
+			if ( shouldRequireSessionToken )
+			{
 			// todo: we should find origin endpoints that can verify game tokens so we don't have to rely on stryder for this in case of a ratelimit
-			if ( request.query.token.includes( "&" ) )
-				return { success: false }
+				if ( request.query.token.includes( "&" ) )
+					return { success: false }
```
Comment (abarichello) [inline client/clientauth.js:33]: Also i'm not sure what should I include here

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (Alystrasz) [inline .vscode/launch.json:1]: Be careful not to include your IDE config into version checking (maybe this should be added to .gitignore?)

File: shared/errorcodes.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const errorCodes = {
+	ACCOUNT_NOT_FOUND: {
+		enum: "ACCOUNT_NOT_FOUND",
```
Comment (Alystrasz) [inline shared/errorcodes.js:3]: I don't get the point of this `enum` property, sounds like code duplication to me.
Is it used client-side?

**Comment by emma-miler** [comment]
@abarichello Since i'll be using custom error messages for my pre-join check code, could you let me know if this is planned to be merged (soon), or if i should base my system on something else.

File: shared/errorcodes.js (hunk 3 above)
Comment (emma-miler) [reply shared/errorcodes.js:3]: As bob said, we could just send a localization string ike '#ACCOUNT_NOT_FOUND' to the client and let that deal with it

Comment (abarichello) [reply shared/errorcodes.js:3]: Yes, the plan is to prepend a '#' to this string and then localize it

**Comment by abarichello** [comment]
lol merged this accidentally

File: server/gameserver.js [hunk 4]
```diff
@@ -179,28 +189,67 @@ module.exports = ( fastify, opts, done ) => {
...
+				}
+				else						//i suppose maybe add the brackets here to as upper one works with it. but actually its fine not to i guess.
+				{
+					server[ key ] = request.query[ key ]
+				}
+			}
+
+			return null
+		} )
+>>>>>>> 6b224de (Add enum codes response)
```
[the remark this answers is not in the document]
Comment (abarichello) [reply server/gameserver.js:223]: https://github.com/R2Northstar/NorthstarMasterServer/pull/24#issuecomment-1052417988

**Pull request merged**